node/network/bridge/src/validator_discovery.rs

```diff
@@ -156,7 +156,7 @@ impl<N: Network, AD: AuthorityDiscovery> Service<N, AD> {
 fn extract_peer_ids(multiaddr: impl Iterator<Item = Multiaddr>) -> HashSet<PeerId> {
 	multiaddr
 		.filter_map(|mut addr| match addr.pop() {
-			Some(multiaddr::Protocol::P2p(peer_id)) => Some(peer_id),
+			Some(multiaddr::Protocol::P2p(key)) => PeerId::from_multihash(key).ok(),
```

Comment:
The build seems to be failing here, any chance you know how to quickly fix it, so I can bake this into `versi`. Thanks.

Reply:
Weird, the polkadot companion is happy. What is the error?

Reply:
Companion CI can fail since upstream Substrate needs to merge first. Thats fine.

Reply:
```
[2023-08-07 08:04:17]    --> node/network/bridge/src/validator_discovery.rs:159:66
[2023-08-07 08:04:17]     |
[2023-08-07 08:04:17] 159 |             Some(multiaddr::Protocol::P2p(key)) => PeerId::from_multihash(key).ok(),
[2023-08-07 08:04:17]     |                                                    ---------------------- ^^^ expected `Multihash<64>`, found `PeerId`
[2023-08-07 08:04:17]     |                                                    |
[2023-08-07 08:04:17]     |                                                    arguments to this function are incorrect
```

I guess it was this.

> Companion CI can fail since upstream Substrate needs to merge first. Thats fine.

Ok, cool did not know that. Thanks!